gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFormat.java

```diff
@@ -67,15 +63,24 @@ public void apply(Project project) {
                 if (eclipseFormattingEnabled(project) && palantirJavaFormatterEnabled(project)) {
                     throw new GradleException(
                             "Can't use both eclipse and palantir-java-format at the same time, please delete one of "
-                                    + ECLIPSE_FORMATTING + " or " + PJF_PROPERTY + " from your gradle.properties");
+                                    + ECLIPSE_FORMATTING
+                                    + " or "
+                                    + PJF_PROPERTY
+                                    + " from your gradle.properties");
                 }
 
                 if (eclipseFormattingEnabled(project)) {
                     java.eclipse().configFile(project.file(eclipseXml.toString()));
                 }
 
                 if (palantirJavaFormatterEnabled(project)) {
-                    java.customLazy("palantir-java-format", PalantirJavaFormatterFunc::new);
+                    Preconditions.checkState(
```

Comment:
who produces `java-format-provider`? This also forces users to apply java-format-provider first. Could we make it perform the logic in a `withPlugin()` block and then verify that it executed with an after evaluate?

Reply:
It's produced in `com.palantir.javaformat:gradle-palantir-java-format`.
This is actually safe for now, as we ensure to apply `java-format-provider` in the main Baseline plugin.